onnxruntime/contrib_ops/cuda/transformers/beam_search.cc

```diff
@@ -48,6 +46,10 @@ BeamSearch::BeamSearch(const OpKernelInfo& info)
                    GenerationCudaDeviceHelper::InitBeamState<float>,
                    GenerationCudaDeviceHelper::InitBeamState<MLFloat16>);
 
+#ifndef USE_ROCM
+  SetDeviceHelpers_Cuda(GenerationCudaDeviceHelper::ReorderPastState, GenerationCudaDeviceHelper::InitCacheIndir);
+#endif
```

Comment:
Should this be `#ifdef USE_CUDA` instead? The `SetDeviceHelpers_Cuda` method in `onnxruntime/contrib_ops/cpu/transformers/beam_search.h` is using `#ifdef USE_CUDA`:

```
#ifdef USE_CUDA
  void SetDeviceHelpers_Cuda(
      const GenerationDeviceHelper::ReorderPastStateFunc& reorder_past_state_func,
      const GenerationDeviceHelper::InitCacheIndirFunc& init_cache_indir_func) {
    reorder_past_state_func_ = reorder_past_state_func;
    init_cache_indir_func_ = init_cache_indir_func;
  }
#endif
```

Reply:
No, this should not be `#ifdef USE_CUDA`, the file exists in `.../cuda/...`, the only case it is not compiling for CUDA is that it is hipified.

For where it is `#ifdef USE_CUDA`, you should notice the file path is `.../cpu/...`